File: atividade2/main.java
package atividade2;

import java.util.ArrayList;
import java.util.Scanner;
import java.util.List;
public class main {
    static boolean  ifAssistido;
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        List<Filme> ListaDeFilmes = new ArrayList<Filme>();
        while (true) {
            System.out.println("Options:");
            System.out.println("1 - Adicionar Filme");
            System.out.println("2 - Marcar um filme como assistido");
            System.out.println("3 - Avaliar um filme");
            System.out.println("4 - Mostrar a lista de filmes");
            System.out.println("5 - Quitar do programa");
            System.out.print("Escolha: ");
            int escolha = input.nextInt();

            switch (escolha) {
                case 1:
                    System.out.print("Entre com o titulo do filme: ");
                    String titulo = input.next();
                    System.out.print("Entre com o ano do filme: ");
                    int ano = input.nextInt();
                    System.out.print("Entre com o genero do filme: ");
                    String genero = input.next();
                    System.out.print("Entre com o duracao do filme: ");
                    String duracao = input.next();
                    System.out.print("Marcar o filme como assistido? (true para sim e false para nao): ");
                    boolean assistido = input.nextBoolean();
                    if(ifAssistido){
                        System.out.println("Avalie o filme: ");
                        float avaliacao = input.nextFloat();
                        ListaDeFilmes.add(new Filme(titulo, ano, genero, duracao, assistido, avaliacao));
                    }else
                        ListaDeFilmes.add(new Filme(titulo, ano, genero, duracao, assistido, 0));
                    break;
                case 2:
                    System.out.print("Marcar filme como assistido: ");
                    titulo = input.next();
                    for(Filme filme: ListaDeFilmes){
                        if( titulo.equals(filme.titulo)) {
                            filme.FoiAssistido(titulo);
                            ifAssistido = true;
                        }
                    }
                    if(ifAssistido){
                        System.out.println("Filme marcado como assistido!");
                        ifAssistido = false;
                    }else{
                        System.out.println("Filme nao encontrado!");
                    }
                    break;
                case 3:
                    System.out.print("Qual filme voce quer avaliar: ");
                    titulo = input.next();
                    for(Filme filme: ListaDeFilmes){
                        if(titulo.equals(filme.titulo)) {
                            System.out.print("Entre com a nota do filme: ");
                            float nota = input.nextFloat();
                            filme.avaliacaoFilme(nota);
                            ifAssistido = true;
                        }
                    }
                    if(ifAssistido){
                        ifAssistido = false;
                    }
                    else{
                        System.out.println("Filme nao encontrado!");
                    }

                    break;
                case 4:
                    for(Filme filme : ListaDeFilmes){
                        filme.ExibirAtributos();
                    }
                    break;
                case 5:
                    System.out.println("adeus!");
                    input.close();
                    System.exit(0);
                    break;
                default:
                    System.out.println("Opção invalida!");
                    break;
            }
        }
    }
}

File: atividade2/main.js
class Filme {
    constructor(titulo, ano, genero, duracao, assistido, avaliacao) {
        this.titulo = titulo;
        this.ano = ano;
        this.genero = genero;
        this.duracao = duracao;
        this.assistido = assistido;
        this.avaliacao = avaliacao;
    }

    FoiAssistido() {
        this.assistido = true;
    }

    avaliacaoFilme(nota) {
        this.avaliacao = nota;
    }

    ExibirAtributos() {
        console.log(`Título: ${this.titulo}`);
        console.log(`Ano: ${this.ano}`);
        console.log(`Gênero: ${this.genero}`);
        console.log(`Duração: ${this.duracao}`);
        console.log(`Assistido: ${this.assistido}`);
        console.log(`Avaliação: ${this.avaliacao}`);
    }
}

const readline = require('readline');
const input = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let ifAssistido = false;
const ListaDeFilmes = [];

function main() {
    console.log("Options:");
    console.log("1 - Adicionar Filme");
    console.log("2 - Marcar um filme como assistido");
    console.log("3 - Avaliar um filme");
    console.log("4 - Mostrar a lista de filmes");
    console.log("5 - Quitar do programa");
    input.question("Escolha: ", (escolha) => {
        escolha = parseInt(escolha);
        switch (escolha) {
            case 1:
                input.question("Entre com o título do filme: ", (titulo) => {
                    input.question("Entre com o ano do filme: ", (ano) => {
                        input.question("Entre com o gênero do filme: ", (genero) => {
                            input.question("Entre com a duração do filme: ", (duracao) => {
                                input.question("Marcar o filme como assistido? (true para sim e false para não): ", (assistido) => {
                                    assistido = (assistido === 'true');
                                    if (ifAssistido) {
                                        input.question("Avalie o filme: ", (avaliacao) => {
                                            ListaDeFilmes.push(new Filme(titulo, ano, genero, duracao, assistido, parseFloat(avaliacao)));
                                            main();
                                        });
                                    } else {
                                        ListaDeFilmes.push(new Filme(titulo, ano, genero, duracao, assistido, 0));
                                        main();
                                    }
                                });
                            });
                        });
                    });
                });
                break;
            case 2:
                input.question("Marcar filme como assistido: ", (titulo) => {
                    for (const filme of ListaDeFilmes) {
                        if (titulo === filme.titulo) {
                            filme.FoiAssistido();
                            ifAssistido = true;
                        }
                    }
                    if (ifAssistido) {
                        console.log("Filme marcado como assistido!");
                        ifAssistido = false;
                    } else {
                        console.log("Filme não encontrado!");
                    }
                    main();
                });
                break;
            case 3:
                input.question("Qual filme você quer avaliar: ", (titulo) => {
                    for (const filme of ListaDeFilmes) {
                        if (titulo === filme.titulo) {
                            input.question("Entre com a nota do filme: ", (nota) => {
                                filme.avaliacaoFilme(parseFloat(nota));
                                ifAssistido = true;
                                main();
                            });
                        }
                    }
                    if (ifAssistido) {
                        ifAssistido = false;
                    } else {
                        console.log("Filme não encontrado!");
                        main();
                    }
                });
                break;
            case 4:
                for (const filme of ListaDeFilmes) {
                    filme.ExibirAtributos();
                }
                main();
                break;
            case 5:
                console.log("adeus!");
                input.close();
                process.exit(0);
                break;
            default:
                console.log("Opção inválida!");
                main();
                break;
        }
    });
}

main();
